src/zh/yidan/src/eu/kanade/tachiyomi/extension/zh/yidan/Yidan.kt

```diff
@@ -1,135 +1,347 @@
+    )
 
     override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
 
+    private fun searchByKeyword(page: Int, query: String): Request {
+        return POST(
+            "$apiUrl/api/searchNovel",
+            headers,
+            KeywordSearchRequest(query).toJsonRequestBody(),
+        )
+    }
+
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val url = "$baseUrl/prod-api/app-api/vv/mh-list/page".toHttpUrl().newBuilder()
-            .apply { if (query.isNotBlank()) addQueryParameter("word", query) }
-            .apply { parseFilters(filters, this) }
-            .addEncodedQueryParameter("pageSize", "50")
-            .addEncodedQueryParameter("pageNo", page.toString())
-            .build()
-        return Request.Builder().url(url).headers(headers).build()
+        if (query.isNotEmpty()) {
+            return searchByKeyword(page, query)
+        }
+        return POST(
+            "$apiUrl/api/getByComicCategoryId",
+            headers,
+            FilterRequest(
+                page = page,
+                limit = PAGE_SIZE,
+                categoryId = filters.firstInstance<CategoryFilter>().selected,
+                orderType = filters.firstInstance<SortFilter>().selected,
+                overType = filters.firstInstance<StatusFilter>().selected,
+            ).toJsonRequestBody(),
+        )
     }
 
-    override fun searchMangaParse(response: Response) = popularMangaParse(response)
+    override fun searchMangaParse(response: Response): MangasPage {
+        val searchByKeyword = response.request.url.toString().contains("searchNovel")
+        val records = if (searchByKeyword) {
+            response.parseAs<CommonResponse<List<Record>>>().result
+        } else {
+            response.parseAs<CommonResponse<FilterResult>>().result.list
+        }
+        return createMangasPage(records, paginated = !searchByKeyword)
+    }
 
-    // for WebView
-    override fun mangaDetailsRequest(manga: SManga) =
-        GET("$baseUrl/#/pages/detail/detail?id=${manga.url}")
+    private fun createMangasPage(records: List<Record>, paginated: Boolean = true): MangasPage {
+        return MangasPage(
+            records.map {
+                SManga.create().apply {
+                    url = "${it.id}"
+                    title = it.novelTitle
+                    thumbnail_url = it.imgUrl
+                }
+            },
+            if (paginated) {
+                records.size >= PAGE_SIZE
+            } else {
+                false
+            },
+        )
+    }
 
-    override fun fetchMangaDetails(manga: SManga): Observable<SManga> {
-        val request = GET("$baseUrl/prod-api/app-api/vv/mh-list/get?id=${manga.url}", headers)
-        return client.newCall(request).asObservableSuccess().map { mangaDetailsParse(it) }
+    override fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl/pages/comic/info?id=${manga.url}"
```

Comment:
Consider:
```suggestion
        return "$baseUrl/pages/comic/info".toHttpUrl().newBuilder()
            .addQueryParameter("id", manga.url)
            .toString()
```

whitespace might be off.